enhancements/installer/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,434 @@
+
+Booting BMs in a virtual network requires specific hardware; this case is a
+simplified one and is not expected to be the most represented in real-world
+installations.
+
+Prerequisites for the user:
+
+* Being able to provision BM nodes on a tenant network
+
+Expected work:
+
+* Adapt the wait-for install-complete timeout, either in code or in
+  documentation, in order to accomodate for the BM booting times.
+* Document how to use:
+  * document the requirements (specific hardware mentioned above)
+  * set the Ironic flavor in `install-config.yaml`
+
+#### Case 2. Install with BM workers on a preexisting network
+
+This case leverages [Provider Networks and Custom
+Subnets](openstack-customer-provided-subnets.md). The cluster is installed on
+the same pre-existing network where the BMs boot.
+
+Prerequisites for the user:
+
+* Being able to provision VMs nodes on the preexisting subnet where BMs run.
+
+Expected work:
+
+* Adapt the wait-for install-complete timeout, either in code or in
+  documentation, in order to accomodate for the BM booting times.
+* Adapt the Certificate-Signing-Request timeouts to the BM booting times.
+* Document how to use:
+  * set the appropriate Ironic-backed Nova flavor in `install-config.yaml`
+
+#### Case 3. Install with both VM and BM workers (BMs on a preexisting network)
+
+This case leverages [Provider Networks and Custom
+Subnets](openstack-customer-provided-subnets.md). The cluster is installed on
+the same pre-existing network where the BMs boot.
+
+Prerequisites for the user:
+
+* Being able to provision VMs nodes on the preexisting subnet where BMs run.
+
+Expected work:
+
+* Enable `install-config.yaml` to allow more than one worker set.
+* Adapt the wait-for install-complete timeout, either in code or in
+  documentation, in order to accomodate for the BM booting times.
+* Adapt the Certificate-Signing-Request timeouts to the BM booting times.
+* Document how to use:
+  * specify two different worker sets in `install-config.yaml`: one for the VM
+    MachineSet, and one with Ironic-backed Nova flavors for the BM MachineSet
+
+#### Case 4. Install on VMs then add BM workers (BMs on a preexisting network)
+
+In this case, the cluster is installed on the installer-provisioned network.
+The additional MachineSet bears the `subnetName` of the preexisting network
+onto which the BMs will boot.
+
+Prerequisites for the user:
+
+* Connectivity must be ensured between the preexisting subnet and the cluster's
+  subnet (eg through a router).
+
+Expected work:
+
+* Allow configuring a Neutron subnet for a given worker node pool.
+* Adapt the Certificate-Signing-Request timeouts to the BM booting times.
+* Document how to use:
+  * make sure the `install-config.yaml` `MachineCIDR` contains the BMs network
+  * add a MachineSet with an Ironic-backed Nova flavor and the correct
+    `subnetName`.
+
+#### Adapt the wait-for install-complete timeout
+
```

Comment:
Need to write a clearer statement of what is the desired functionality: 
" The installer should not incorrectly time out waiting for BMs workers to boot up"  (??) 
or
" Installer should be modified to accommodate longer boot times for BM workers, so installation does not incorrectly timeout"

Reply:
Thanks; I have changed the text to be more explicit. Can you have a look?